File: CSharp/tests~/Assets/Tests/Runtime/Scripts/XLog/tag_test.cs
// Copyright (c) 2026 FrameworkEase Technologies. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

using System.Collections.Generic;
using FrameworkEase.Universal.X.Utility;
using NUnit.Framework;

public partial class TestXLog
{
    [Test]
    public void TagCount()
    {
        var tag = new XLog.Tag();
        Assert.That(tag.Count, Is.EqualTo(0));

        tag.Set("key1", "value1");
        Assert.That(tag.Count, Is.EqualTo(1));
    }

    [Test]
    public void TagSet()
    {
        var tag = new XLog.Tag();
        tag.Set("key1", "value1");
        Assert.That(tag.Get("key1"), Is.EqualTo("value1"));

        tag.Set("key2", "value2");
        Assert.That(tag.Get("key2"), Is.EqualTo("value2"));

        tag.Set("key1", "value1_updated");
        Assert.That(tag.Get("key1"), Is.EqualTo("value1_updated"));
    }

    [Test]
    public void TagGet()
    {
        var tag = new XLog.Tag();
        Assert.That(tag.Get("nonexistent"), Is.EqualTo(""));

        tag.Set("key1", "value1");
        Assert.That(tag.Get("key1"), Is.EqualTo("value1"));
    }

    [Test]
    public void TagRange()
    {
        var tag = new XLog.Tag();
        tag.Set("key1", "value1");
        tag.Set("key2", "value2");
        tag.Set("key3", "value3");
        var pairs = new List<string>();
        tag.Range((key, value) =>
        {
            pairs.Add(key);
            pairs.Add(value);
            if (key == "key2") return false;
            else return true;
        });
        Assert.That(pairs, Is.EqualTo(new[] { "key1", "value1", "key2", "value2" }));
    }

    [Test]
    public void TagStringify()
    {
        var tag = new XLog.Tag();
        Assert.That(tag.Stringify(), Is.EqualTo(""));

        tag.Set("key1", "value1");
        Assert.That(tag.Stringify(), Is.EqualTo("[key1=value1]"));

        tag.Set("key2", "value2");
        Assert.That(tag.Stringify(), Is.EqualTo("[key1=value1, key2=value2]"));
    }
}

File: Go/XLog/tag_test.go
// Copyright (c) 2026 FrameworkEase Technologies. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

package XLog

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestTagCount(t *testing.T) {
	tag := Tag{}
	assert.Equal(t, 0, tag.Count())

	tag.Set("key1", "value1")
	assert.Equal(t, 1, tag.Count())
}

func TestTagSet(t *testing.T) {
	tag := Tag{}
	tag.Set("key1", "value1")
	assert.Equal(t, "value1", tag.Get("key1"))

	tag.Set("key2", "value2")
	assert.Equal(t, "value2", tag.Get("key2"))

	tag.Set("key1", "value1_updated")
	assert.Equal(t, "value1_updated", tag.Get("key1"))
}

func TestTagGet(t *testing.T) {
	tag := Tag{}
	assert.Equal(t, "", tag.Get("nonexistent"))

	tag.Set("key1", "value1")
	assert.Equal(t, "value1", tag.Get("key1"))
}

func TestTagRange(t *testing.T) {
	tag := Tag{}
	tag.Set("key1", "value1")
	tag.Set("key2", "value2")
	tag.Set("key3", "value3")
	var pairs []string
	tag.Range(func(key string, value string) bool {
		pairs = append(pairs, key, value)
		if key == "key2" {
			return false
		}
		return true
	})
	assert.Equal(t, []string{"key1", "value1", "key2", "value2"}, pairs)
}

func TestTagStringify(t *testing.T) {
	tag := Tag{}
	assert.Equal(t, "", tag.Stringify())

	tag.Set("key1", "value1")
	assert.Equal(t, "[key1=value1]", tag.Stringify())

	tag.Set("key2", "value2")
	assert.Equal(t, "[key1=value1, key2=value2]", tag.Stringify())
}
